rivus/gridder/extend_grid.py

```diff
@@ -6,21 +6,52 @@
 # Add Commodity columns filled with to the vertex DataFrame,
 def vert_init_commodities(vertex_df, commodities, sources=None):
     """Add commodity columns with zeros to vertex DataFrame
-        source: (Commodity, Index, Value) tuples/lists
-            e.g. ('Elec', 1, 2000)
+    source: (Commodity, Index, Value) tuples/lists
+        e.g. ('Elec', 1, 2000)
+
+    Args:
+        vertex_df ((Geo)DataFrame): vertex dataframe input
+        commodities (lis of str): Like ('Elec', 'Gas', 'Heat')
+        sources (list of tuples, optional): Init the source nodes
+            Like [('Elec', 0, 100000), ('Gas', 1, 50000)]
     """
     for commo in commodities:
         vertex_df[commo] = len(vertex_df) * [0]
     if sources:
         for s in sources:
             check1 = (len(s) == 3) and isinstance(s[0], str)
-            check2 = isinstance(s[1], int) and isinstance(s[2], (int, float))
+            check2 = isinstance(s[1], int) and isinstance(s[2], (int, float)) and \
+                s[1] < len(vertex_df)
             if check1 and check2:
                 vertex_df.set_value(index=s[1], col=s[0], value=s[2])
+            else:
+                raise ValueError('Parameter problem in function call. ch1: {}, ch2: {}'.format(check1, check2))
 
 
 def extend_edge_data(edge_df, dtype='area', strat='equal', sorts=None, inits=None):
-    """add various data to the edge (Geo)DataFrame"""
+    """Add various data to the edge (Geo)DataFrame
+        Normally used to add demand data
+    
+    Parameters
+    ----------
+    edge_df : (Geo)DataFrame
+        edge dataframe to be extended
+    dtype : str, optional
+        Type of data whit wich we extend (todo)
+    strat : str, optional
```

Comment:
List possible values and a one-line description of each somewhere in the docstring.